packages/astro/src/runtime/server/render/astro/instance.ts

```diff
@@ -31,8 +31,9 @@ export class AstroComponentInstance {
 		this.factory = factory;
 		this.slotValues = {};
 		for (const name in slots) {
-			const value = slots[name](result);
-			this.slotValues[name] = () => value;
+			// collections-renderentry.test.js this units needs to early get the css from the children component and the slot  
+			slots[name](result)
+			this.slotValues[name] = () => slots[name](result);
```

Comment:
Calling this twice feels wrong to me. I can believe it fixes the test but it's puzzling why that would be.

Reply:
I have tried a lot but I still haven’t found the way it broke that unit test.
It's very weird that it only breaks that one-unit test

Reply:
https://github.com/withastro/astro/blob/31cbf4357e866abab62bb4ccf949f5a59aec06ac/packages/astro/test/units/dev/collections-renderentry.test.js#L157
There isn’t CSS, but the CSS style tag even appears in the bundle.

Reply:
@matthewp coule help me check this